library/src/main/java/com/bumptech/glide/load/resource/bitmap/CenterCrop.java

```diff
@@ -13,24 +13,16 @@
  *
  * Does not maintain the image's aspect ratio
  */
-public class CenterCrop extends BitmapTransformation {
+public class CenterCrop extends PositionedCrop {
   private static final String ID = "com.bumptech.glide.load.resource.bitmap.CenterCrop";
   private static final byte[] ID_BYTES = ID.getBytes(CHARSET);
 
   public CenterCrop(Context context) {
-    super(context);
+    super(context, 0.5f, 0.5f);
```

Comment:
Hmm, good idea; I was only thinking locally in CenterCrop.

I would expand to left/right, top/bottom, and maybe center/middle, because start-end is misleading. There's no reason to abstract it away, especially since Gravity already has these start/end names, which are direction-sensitive, but these constants are not that!

In the future (in a later PR or if you have a lot of time) it would be nice to have a specialized constructor that accepts a Gravity `int` and decompose that to the corresponding float values.